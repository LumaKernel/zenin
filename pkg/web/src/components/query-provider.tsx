"use client";

import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import { ReactQueryDevtools } from "@tanstack/react-query-devtools";

type QueryProviderProps = {
  readonly children: React.ReactNode;
};

const queryClient = new QueryClient();

export function QueryProvider({ children }: QueryProviderProps) {
  return (
    <QueryClientProvider client={queryClient}>
      {children}
      <ReactQueryDevtools initialIsOpen={false} />
    </QueryClientProvider>
  );
}