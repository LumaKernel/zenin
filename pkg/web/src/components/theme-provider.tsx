"use client";

import { MantineColorScheme, useMantineColorScheme } from "@mantine/core";
import { useEffect } from "react";

interface ThemeProviderProps {
  readonly children: React.ReactNode;
}

export function ThemeProvider({ children }: ThemeProviderProps) {
  const { setColorScheme } = useMantineColorScheme();

  useEffect(() => {
    const storedTheme = localStorage.getItem("color-scheme") as MantineColorScheme | null;
    if (storedTheme) {
      setColorScheme(storedTheme);
    } else {
      setColorScheme("auto");
    }
  }, [setColorScheme]);

  return <>{children}</>;
}

export function useTheme() {
  const { colorScheme, setColorScheme } = useMantineColorScheme();

  const toggleColorScheme = (scheme: MantineColorScheme) => {
    setColorScheme(scheme);
    localStorage.setItem("color-scheme", scheme);
  };

  return {
    colorScheme,
    setColorScheme: toggleColorScheme,
  };
}