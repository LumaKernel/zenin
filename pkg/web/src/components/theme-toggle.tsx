"use client";

import { ActionIcon, Menu, Text, useMantineColorScheme } from "@mantine/core";
import { IconMoon, IconSun, IconDeviceDesktop } from "@tabler/icons-react";

export function ThemeToggle() {
  const { colorScheme, setColorScheme } = useMantineColorScheme();

  return (
    <Menu shadow="md" width={200}>
      <Menu.Target>
        <ActionIcon variant="outline" size="lg">
          {colorScheme === "dark" ? (
            <IconMoon size={18} />
          ) : colorScheme === "light" ? (
            <IconSun size={18} />
          ) : (
            <IconDeviceDesktop size={18} />
          )}
        </ActionIcon>
      </Menu.Target>

      <Menu.Dropdown>
        <Menu.Label>テーマを選択</Menu.Label>

        <Menu.Item
          leftSection={<IconSun size={14} />}
          onClick={() => setColorScheme("light")}
        >
          <Text size="sm">ライト</Text>
        </Menu.Item>

        <Menu.Item
          leftSection={<IconMoon size={14} />}
          onClick={() => setColorScheme("dark")}
        >
          <Text size="sm">ダーク</Text>
        </Menu.Item>

        <Menu.Item
          leftSection={<IconDeviceDesktop size={14} />}
          onClick={() => setColorScheme("auto")}
        >
          <Text size="sm">システム</Text>
        </Menu.Item>
      </Menu.Dropdown>
    </Menu>
  );
}
