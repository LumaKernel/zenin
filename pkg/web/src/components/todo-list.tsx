"use client";

import {
  Stack,
  Text,
  Card,
  Group,
  ActionIcon,
  TextInput,
  Button,
  Checkbox,
  Alert,
  Loader,
} from "@mantine/core";
import { IconTrash, IconPlus, IconAlertCircle } from "@tabler/icons-react";
import { useState } from "react";
import { useEffectQuery, useEffectMutation } from "../hooks/use-effect-query";
import { useApi } from "../contexts/api-context";
import { Todo, TodoId } from "@domain/TodosApi";

export function TodoList() {
  const [newTodoText, setNewTodoText] = useState("");
  const { todosApi } = useApi();

  const {
    data: todos,
    isLoading,
    error,
  } = useEffectQuery(["todos"], todosApi.getAllTodos());

  const createTodoMutation = useEffectMutation(
    (text: string) => todosApi.createTodo(text),
    {
      onSuccess: () => {
        setNewTodoText("");
      },
      invalidateQueries: [["todos"]],
    },
  );

  const completeTodoMutation = useEffectMutation(
    (id: TodoId) => todosApi.completeTodo(id),
    {
      invalidateQueries: [["todos"]],
    },
  );

  const removeTodoMutation = useEffectMutation(
    (id: TodoId) => todosApi.removeTodo(id),
    {
      invalidateQueries: [["todos"]],
    },
  );

  const handleCreateTodo = () => {
    if (newTodoText.trim()) {
      createTodoMutation.mutate(newTodoText.trim());
    }
  };

  const handleToggleTodo = (todo: Todo) => {
    if (!todo.done) {
      completeTodoMutation.mutate(todo.id);
    }
  };

  const handleRemoveTodo = (id: TodoId) => {
    removeTodoMutation.mutate(id);
  };

  if (isLoading) {
    return (
      <Stack align="center" p="xl">
        <Loader />
        <Text>Todoを読み込み中...</Text>
      </Stack>
    );
  }

  if (error) {
    return (
      <Alert icon={<IconAlertCircle size="1rem" />} title="エラー" color="red">
        Todoの取得に失敗しました: {String(error)}
      </Alert>
    );
  }

  return (
    <Stack>
      <Text size="xl" fw={700}>
        Todo一覧
      </Text>

      <Card p="md">
        <Group>
          <TextInput
            placeholder="新しいTodoを入力"
            value={newTodoText}
            onChange={(event) => setNewTodoText(event.currentTarget.value)}
            style={{ flex: 1 }}
            onKeyPress={(event) => {
              if (event.key === "Enter") {
                handleCreateTodo();
              }
            }}
          />
          <Button
            leftSection={<IconPlus size="1rem" />}
            onClick={handleCreateTodo}
            loading={createTodoMutation.isPending}
            disabled={!newTodoText.trim()}
          >
            追加
          </Button>
        </Group>
      </Card>

      <Stack gap="xs">
        {todos?.map((todo) => (
          <Card key={todo.id} p="md">
            <Group justify="space-between">
              <Group>
                <Checkbox
                  checked={todo.done}
                  onChange={() => handleToggleTodo(todo)}
                  disabled={completeTodoMutation.isPending}
                />
                <Text
                  style={{
                    textDecoration: todo.done ? "line-through" : "none",
                    opacity: todo.done ? 0.6 : 1,
                  }}
                >
                  {todo.text}
                </Text>
              </Group>
              <ActionIcon
                color="red"
                variant="light"
                onClick={() => handleRemoveTodo(todo.id)}
                loading={removeTodoMutation.isPending}
              >
                <IconTrash size="1rem" />
              </ActionIcon>
            </Group>
          </Card>
        ))}
        {todos?.length === 0 && (
          <Text c="dimmed" ta="center" p="xl">
            Todoはありません
          </Text>
        )}
      </Stack>
    </Stack>
  );
}
