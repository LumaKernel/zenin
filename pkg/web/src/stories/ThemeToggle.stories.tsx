import type { Meta, StoryObj } from "@storybook/react";
import { ThemeToggle } from "../components/theme-toggle";

const meta = {
  title: "Components/ThemeToggle",
  component: ThemeToggle,
  parameters: {
    layout: "centered",
  },
  tags: ["autodocs"],
} satisfies Meta<typeof ThemeToggle>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {};

export const LightTheme: Story = {
  parameters: {
    backgrounds: { default: 'light' },
  },
};

export const DarkTheme: Story = {
  parameters: {
    backgrounds: { default: 'dark' },
  },
};