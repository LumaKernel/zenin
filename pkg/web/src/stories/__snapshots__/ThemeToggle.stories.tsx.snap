// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/ThemeToggle DarkTheme smoke-test 1`] = `
<style data-mantine-styles="classes">
  @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
</style>
<button class="mantine-focus-auto mantine-active m_8d3f4000 mantine-ActionIcon-root m_87cf2631 mantine-UnstyledButton-root"
        data-variant="outline"
        data-size="lg"
        type="button"
        aria-haspopup="menu"
        aria-expanded="false"
        aria-controls="mantine-c27oyynwd-dropdown"
        id="mantine-c27oyynwd-target"
        style="--ai-size: var(--ai-size-lg); --ai-bg: transparent; --ai-hover: var(--mantine-color-blue-outline-hover); --ai-color: var(--mantine-color-blue-outline); --ai-bd: calc(0.0625rem * var(--mantine-scale)) solid var(--mantine-color-blue-outline);"
>
  <span class="m_8d3afb97 mantine-ActionIcon-icon">
    <svg xmlns="http://www.w3.org/2000/svg"
         width="18"
         height="18"
         viewbox="0 0 24 24"
         fill="none"
         stroke="currentColor"
         stroke-width="2"
         stroke-linecap="round"
         stroke-linejoin="round"
         class="tabler-icon tabler-icon-sun "
    >
      <path d="M12 12m-4 0a4 4 0 1 0 8 0a4 4 0 1 0 -8 0">
      </path>
      <path d="M3 12h1m8 -9v1m8 8h1m-9 8v1m-6.4 -15.4l.7 .7m12.1 -.7l-.7 .7m0 11.4l.7 .7m-12.1 -.7l-.7 .7">
      </path>
    </svg>
  </span>
</button>
`;

exports[`Components/ThemeToggle Default smoke-test 1`] = `
<style data-mantine-styles="classes">
  @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
</style>
<button class="mantine-focus-auto mantine-active m_8d3f4000 mantine-ActionIcon-root m_87cf2631 mantine-UnstyledButton-root"
        data-variant="outline"
        data-size="lg"
        type="button"
        aria-haspopup="menu"
        aria-expanded="false"
        aria-controls="mantine-d6z4mox8s-dropdown"
        id="mantine-d6z4mox8s-target"
        style="--ai-size: var(--ai-size-lg); --ai-bg: transparent; --ai-hover: var(--mantine-color-blue-outline-hover); --ai-color: var(--mantine-color-blue-outline); --ai-bd: calc(0.0625rem * var(--mantine-scale)) solid var(--mantine-color-blue-outline);"
>
  <span class="m_8d3afb97 mantine-ActionIcon-icon">
    <svg xmlns="http://www.w3.org/2000/svg"
         width="18"
         height="18"
         viewbox="0 0 24 24"
         fill="none"
         stroke="currentColor"
         stroke-width="2"
         stroke-linecap="round"
         stroke-linejoin="round"
         class="tabler-icon tabler-icon-sun "
    >
      <path d="M12 12m-4 0a4 4 0 1 0 8 0a4 4 0 1 0 -8 0">
      </path>
      <path d="M3 12h1m8 -9v1m8 8h1m-9 8v1m-6.4 -15.4l.7 .7m12.1 -.7l-.7 .7m0 11.4l.7 .7m-12.1 -.7l-.7 .7">
      </path>
    </svg>
  </span>
</button>
`;

exports[`Components/ThemeToggle LightTheme smoke-test 1`] = `
<style data-mantine-styles="classes">
  @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
</style>
<button class="mantine-focus-auto mantine-active m_8d3f4000 mantine-ActionIcon-root m_87cf2631 mantine-UnstyledButton-root"
        data-variant="outline"
        data-size="lg"
        type="button"
        aria-haspopup="menu"
        aria-expanded="false"
        aria-controls="mantine-k9dl85wvs-dropdown"
        id="mantine-k9dl85wvs-target"
        style="--ai-size: var(--ai-size-lg); --ai-bg: transparent; --ai-hover: var(--mantine-color-blue-outline-hover); --ai-color: var(--mantine-color-blue-outline); --ai-bd: calc(0.0625rem * var(--mantine-scale)) solid var(--mantine-color-blue-outline);"
>
  <span class="m_8d3afb97 mantine-ActionIcon-icon">
    <svg xmlns="http://www.w3.org/2000/svg"
         width="18"
         height="18"
         viewbox="0 0 24 24"
         fill="none"
         stroke="currentColor"
         stroke-width="2"
         stroke-linecap="round"
         stroke-linejoin="round"
         class="tabler-icon tabler-icon-sun "
    >
      <path d="M12 12m-4 0a4 4 0 1 0 8 0a4 4 0 1 0 -8 0">
      </path>
      <path d="M3 12h1m8 -9v1m8 8h1m-9 8v1m-6.4 -15.4l.7 .7m12.1 -.7l-.7 .7m0 11.4l.7 .7m-12.1 -.7l-.7 .7">
      </path>
    </svg>
  </span>
</button>
`;
